File: 프로그래머스/lv2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;



vector<int> solution(int brown, int yellow) {
    vector<int> answer;

    int row=1;
    int col;
    int cnt=0;
    
    while(1){
        if(yellow%row==0){
            col=yellow/row;
            if(row*2+col*2+4==brown){
                answer.push_back(col+2);
                answer.push_back(row+2);
                return answer;
            } 
        }
        row++;
    }
    
}

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        
        int row = 1;
        
        // 갈색 = (width+height-2) x 2 
        // 노란색 = width x height - [ (width+height-2) x 2 ]
        
        while(true){

            if(yellow%row!=0){
                row++;
                continue;
            }
            
            int num = row*2 + ((yellow/row)+2)*2;
            if(num == brown){
                return new int[]{(yellow/row)+2,row+2};
            }
            row++;
        }
        
    }
}